homeassistant/components/notify/pushover.py

```diff
@@ -64,13 +89,87 @@ def send_message(self, message='', **kwargs):
         if not isinstance(targets, list):
             targets = [targets]
 
+        file = {}
+
+        if file_data is not None:
+            filename = self.load_file(
+                url=file_data.get(ATTR_FILE_URL),
+                local_path=file_data.get(ATTR_FILE_PATH),
+                username=file_data.get(ATTR_FILE_USERNAME),
+                password=file_data.get(ATTR_FILE_PASSWORD),
+                auth=file_data.get(ATTR_FILE_AUTH))
+
+            if filename is not None:
+                file = (filename, open(
+                    filename, "rb"), mimetypes.guess_type(filename))
+
         for target in targets:
             if target is not None:
                 data['device'] = target
 
             try:
-                self.pushover.send_message(message, **data)
+                self.pushover.send_message(
+                    message=message, attachment=file, **data)
+
             except ValueError as val_err:
                 _LOGGER.error(str(val_err))
             except RequestError:
                 _LOGGER.exception("Could not send pushover notification")
+            finally:
+                # Remove the tempfile if one was created
+                if file_data and file_data.get(ATTR_FILE_URL) and filename:
+                    os.remove(filename)
+
+    def load_file(self, url=None, local_path=None, username=None,
+                  password=None, auth=None):
+        """Load image/document/etc from a local path or URL."""
+        try:
+            # Load the file from URL
+            if url is not None:
+                # Check whether authentication parameters are provided
+                if username is not None and password is not None:
```

Comment:
need no None check, simple `if username and password:`

Reply:
Your suggestion isn't the same logic.  In your scenario, a blank password (not something I'd recommend, but also not something we should enforce) would not pass through.  Example config that makes the suggested code fail:
`
file:
  url: "http://www.example.com/image.png"
  auth: basic
  username: "testuser"
  password: ""
`